medical/models/medical_patient.py

```diff
@@ -93,22 +94,52 @@ def _create_vals(self, vals):
         vals = super(MedicalPatient, self)._create_vals(vals)
         if not vals.get('identification_code'):
             Seq = self.env['ir.sequence']
-            vals['identification_code'] = Seq.next_by_code(
+            vals['identification_code'] = Seq.sudo().next_by_code(
                 self._name,
             )
         vals.update({
             'customer': True,
         })
         return vals
 
-    @api.model
-    def _get_default_image(self, vals):
-        res = super(MedicalPatient, self)._get_default_image(vals)
-        if res:
-            return res
-        img_path = get_module_resource(
+    @api.model_cr_context
+    def _get_default_image_path(self, vals):
+        super(MedicalPatient, self)._get_default_image_path(vals)
+        return get_module_resource(
             'medical', 'static/src/img', 'patient-avatar.png',
         )
-        with open(img_path, 'r') as image:
-            base64_image = image.read().encode('base64')
-        return tools.image_resize_image_big(base64_image)
+
+    def _search_age(self, operator, value):
```

Comment:
IMO, this treatment is confusing since it leads to incongruence in the user interface. See ex. below:

![image](https://cloud.githubusercontent.com/assets/25898232/26398804/f6bd72da-407a-11e7-85f8-7367db432786.png)

IMO, since the field Age is treated as a character the user can't choose 'greater/less than' operators anyway to create the filter and the only operator is going to be 'like' it would be better to leave the search behaviour as a string and let the user guess that he can use 'contains'-> '67y'. Also, this way allows to do month searches which would be useful for babies.

Maybe,  a related, invisible, integer 'age2' field to add in the search view and allow comparatives? I don't know if operators can be added in the custom filters.